doc/psidk/src/specifications/app-architecture/plugins.md

```diff
@@ -82,30 +82,64 @@ Supervisor->>psinode: Submit transaction
+Interactions with smart-contracts can get complex. It often requires specialized knowledge from the development team to know how to correctly integrate with it. For apps on a psibase network, the development team will have created a plugin that abstracts the complexity of the interactions with the service into a streamlined user-facing API that can be used to much more easily integrate with the service.
 
 For example, consider an application that requires the user to submit a custom zero-knowledge proof in order to call a particular service action. This would ordinarily likely require the use of complicated cryptographic libraries to generate a proof of the correct type and format expected by the service. In the case of a psibase app, the zero-knowledge app would have a plugin used by their own UI that does the proof generation. This plugin would automatically be available for use by other third-party applications to exercise the same functionality for simpler integration.
 
 ### Private user data
 
 Blockchains typically only have public data. If private client-side data is collected, it's siloed in the application that does the collection. Libraries built on top of client-side synchronous calls can be used to create a private client-side data layer for application to leverage private user data.
+
+## Local data storage
+
+### Access to database operations
+
+The psibase plugin host provides a `wasi:keyvalue` interface for plugins to use. However, the wasi-keyvalue interface [proposal](https://github.com/WebAssembly/wasi-keyvalue) is still in stage 2 of standardization, and therefore the implementation provided by the host is subject to change.
+
+For more stable interactions with the database, a plugin should import the `clientdata` plugin, which is a wrapper around the host's `wasi:keyvalue` interface.
+
+This `clientdata` plugin provides a high level key/value API for plugins.
+
+```
+    get: func(key: string) -> option<list<u8>>;
+    set: func(key: string, value: list<u8>) -> result<_, error>;
+    delete: func(key: string);
+```
+
+### Data backing
+
+The `wasi-keyvalue` implementation currently uses LocalStorage as its data backing, because it is a synchronous interface which is easiest to integrate with wasm.
+
+Note: This imposes a major restriction on the amount of memory available for psibase plugins: Currently there is a hard limit of 5MB for the total amount of data stored in LocalStorage across all psibase apps. This restriction will be lifted in a future implementation of plugin storage that uses a different data backing layer such as IndexedDB.
```

Comment:
The limit is browser specific.

Reply:
Removed the number and just linked to a web page about the limit